### ISSUE TITLE
 No such file or directory:"data/csga/linking/csga linked span train.jsonl'

### ISSUE THREAD

**Issue by Chengyui** [yeeeqichen/FiTs#3]
Deer author, I was runing the run_post_train.sh with align_mask=True, but  find No such file or directory:"data/csga/linking/csga linked span train.jsonl'.  So it seems like some jsonl may not correctly commited?

**Comment by CodeBoyPhilo** [comment]
Hi, from a quick glance, I believe these files are created if they do not exist, as fixed in #1.

The relevant code snippets can be found here: https://github.com/yeeeqichen/FiTs/blob/07d6e9a7bed7e9e20f8c5530411a4d39080822f7/utils/data_utils.py#L1139-L1149

However, I think these are the exact lines of code that triggered the error you mentioned. From my quick review, it seems that the intermediate directories are not being created beforehand. As a quick workaround, you may manually create these directories prior to running the code, or you can add the following lines before saving the aligned data as referenced above:

```python
os.makedirs(os.path.dirname(linked_span_save_path), exist_ok=True)
os.makedirs(os.path.dirname(linked_ids_save_path), exist_ok=True)
os.makedirs(os.path.dirname(linking_mask_save_path), exist_ok=True)
```

Best regards!

**Comment by Chengyui** [comment]
Thanks for your detailed answer, my issues solved. Best Regards!!
